test/powershell/engine/Job/Jobs.Tests.ps1

```diff
@@ -1,23 +1,275 @@
+        }
+
+        It 'Can receive-job using ComputerName' {
+            $jobName = 'ReceiveUsingComputerName'
+            $job = Start-Job -ScriptBlock { 1 + 1 } -Name $jobName | Wait-Job
+            $result = Receive-Job -ComputerName localhost -Job $job
+            $result | Should Be 2
+            $job | Remove-Job -Force -ErrorAction SilentlyContinue
+        }
+
+        It 'Can receive-job using Location' {
+            $jobName = 'ReceiveUsingLocation'
+            $job = Start-Job -ScriptBlock { 1 + 1 } -Name $jobName | Wait-Job
+            $result = Receive-Job -Location localhost -Job $job
+            $result | Should Be 2
+            $job | Remove-Job -Force -ErrorAction SilentlyContinue
+        }
+    }
+
+    Context 'Get-Job tests' {
+        BeforeAll {
+            $getJobTestCases = @(
+                @{ parameters = @{ Name = $startedJob.Name } ; property = 'Name'},
+                @{ parameters = @{ Id = $startedJob.Id } ; property = 'Id'},
+                @{ parameters = @{ InstanceId = $startedJob.InstanceId } ; property = 'InstanceId'},
+                @{ parameters = @{ State = $startedJob.State } ; property = 'State'}
+            )
+
+            $getJobSwitches = @(
+                @{ parameters = @{ Before = $timeAfterStartedJob }; property = '-Before'},
+                @{ parameters = @{ After = $timeBeforeStartedJob }; property = '-After'},
+                @{ parameters = @{ HasMoreData = $true }; property = '-HasMoreData'}
+            )
+
+            $getJobChildJobs = @(
+                @{ parameters = @{ IncludeChildJob = $true }; property = '-IncludeChildJob'},
+                @{ parameters = @{ ChildJobState = 'Completed' }; property = '-ChildJobState'}
+            )
+        }
+
+        BeforeEach {
+            Get-Job | Where-Object { $_.Id -ne $startedJob.Id } | Remove-Job -ErrorAction SilentlyContinue -Force
+        }
+
+        It 'Can get-job with <property>' -TestCases $getJobTestCases {
+            param($parameters)
+            $job = Get-Job @parameters
+            ValidateJobInfo -job $job -state 'Completed' -hasMoreData $true -command ' 1 + 1 '
+        }
+
+        It 'Can get-job with <property>' -TestCases $getJobSwitches {
+            param($parameters)
+            $job = Get-Job @parameters
+            ValidateJobInfo -job $job -state 'Completed' -hasMoreData $true -Name 'StartedJob'
+        }
+
+        It 'Can get-job with <property>' -TestCases $getJobChildJobs {
+            param($parameters)
+            $jobs = Get-Job @parameters
+            $jobs.Count | Should Be 2
+            ValidateJobInfo -job $jobs[0] -state 'Completed' -hasMoreData $true -Name 'StartedJob'
+            ValidateJobInfo -job $jobs[1] -state 'Completed' -hasMoreData $true
+        }
+    }
+
+
+    Context 'Remove-Job tests' {
+        # The test pattern used here is different from other tests since there is a scoping issue in Pester.
```

Comment:
Is there an issue for this in the Pester repo? If so, please include a link

Reply:
I need to do a deeper analysis to file an issue. It could be Pester or PowerShell scoping issue.